### PR TITLE
Update Makefile to use CGO_ENABLED environment variable

### PR DESCRIPTION
**Pull request by sukhil-suresh**
There are scenarios in which `CGO_ENABLED=1` has to be set when building the `gomplate` binary. 
Specifically building the binary in FIPS-enabled mode requires that `CGO_ENABLED=1` and `GOEXPERIMENT=boringcrypto` are set.
However, this is not possible with the Makefile hardcoding `CGO_ENABLED=0`. 

This PR makes changes to allow the value of the `CGO_ENABLED` environment variable to be used if it is set. 
If the `CGO_ENABLED` environment variable is not set, then `CGO_ENABLED` defaults to 0.

### CODE REVIEW

**Comment by hairyhenderson** [comment]
Thanks for the contribution, @sukhil-suresh!

File: Makefile [hunk 1]
```diff
@@ -28,6 +28,8 @@ GO_LDFLAGS ?= $(COMMIT_FLAG) $(VERSION_FLAG)
 GOOS ?= $(shell $(GO) version | sed 's/^.*\ \([a-z0-9]*\)\/\([a-z0-9]*\)/\1/')
 GOARCH ?= $(shell $(GO) version | sed 's/^.*\ \([a-z0-9]*\)\/\([a-z0-9]*\)/\2/')
 
+CGO_ENABLED ?= 0
```
Comment (hairyhenderson) [inline Makefile:31]:
```suggestion
# allow overriding CGO_ENABLED for scenarios where gomplate must be compiled with CGO enabled, such as when using boringcrypto.
CGO_ENABLED ?= 0
```

**Review by hairyhenderson — COMMENTED** [review]
LGTM - just need an explanatory comment (I'll definitely forget why we'd ever want to override it!)

File: Makefile (hunk 1 above)
Comment (sukhil-suresh) [reply Makefile:32]: Updated as suggested. Marking this conversation as resolved.

**Comment by sukhil-suresh** [comment]
@hairyhenderson please take another look. I have made the suggested change.

**Review by hairyhenderson — APPROVED** [review]
Thanks @sukhil-suresh!

**Pull request merged**